apps/spruce/src/utils/string/Linkify.tsx

```diff
@@ -0,0 +1,40 @@
+import reactStringReplace from "react-string-replace";
+import { StyledLink } from "components/styles";
+import { getJiraTicketUrl } from "constants/externalResources";
+
+const githubPRLinkify = (unlinkified: string | React.ReactNode[]) =>
+  reactStringReplace(
+    unlinkified,
+    /(https:\/\/github.com\/[a-zA-Z0-9-]+\/[a-zA-Z0-9-]+\/pull\/\d+)/g,
+    (match, i) => (
+      <StyledLink key={`${match}${i}`} href={match}>
+        {match}
+      </StyledLink>
+    ),
+  );
+
+const jiraLinkify = (
+  unlinkified: string | React.ReactNode[],
+  jiraHost: string,
+  // @ts-expect-error: FIXME. This comment was added by an automated script.
+  onClick?,
+) =>
+  reactStringReplace(unlinkified, /([A-Z]{1,10}-\d{1,6})/g, (match, i) => (
+    <StyledLink
+      key={`${match}${i}`}
+      data-cy="jira-link"
+      href={getJiraTicketUrl(jiraHost, match)}
+      onClick={onClick}
+    >
+      {match}
+    </StyledLink>
+  ));
+
+const linkifyString = (unlinkified: string) =>
```

Comment:
It looks like we have a [dependency `linkify-html`](https://github.com/evergreen-ci/ui/blob/7b76bf5af3a83e09c1a77fead48dc945e7abcc58/apps/spruce/src/pages/task/taskTabs/logs/logTypes/LogMessageLine.tsx#L30) that does this.

I wonder if the dependency has more security handling, or if we should just drop it?

Reply:
I think we should just drop it tbh, you have to call the `parse` function on top of `linkify-html` to parse it into real html and I think it's not worth it

I'll keep the other changes for testing the preexisting linkify functions though since we probably should have had tests for those!